### PR TITLE
Issue 155 doi info rework

### LINKED ISSUE #155: DOI Metadata handling rework
### Pacifica Metadata version
all

### Pacifica Core Software versions
current

### Platform Details
[Operating system distribution and release version. Cloud provider if running in the cloud]

### Scenario:
Need new and more complete tables for storing and manipulating metadata for DOI requests and records

### PR DESCRIPTION
**Pull request by kauberry**
### Description

Removing deprecated DOI handling tables, and adding new tables and testing code

### Issues Resolved

#155 

### Check List

- [ ] All tests pass. See <https://github.com/pacifica/pacifica-docs/blob/master/development/TESTING.md>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: metadata/orm/doi_entries.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""Keywords linked to citations."""
```
Comment (dmlb2000) [inline metadata/orm/doi_entries.py]: Fix the comment here...

File: metadata/orm/doi_authors.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+"""Keywords linked to citations."""
+from peewee import CharField, IntegerField, ForeignKeyField
+from metadata.rest.orm import CherryPyAPI
+from metadata.orm.doi_entries import DOIEntries
+from metadata.orm.utils import unicode_type
+
+
+class DOIAuthors(CherryPyAPI):
+    """
+    DOI Entries Model.
```
Comment (dmlb2000) [inline metadata/orm/doi_authors.py]: Fix the comment here...

File: metadata/orm/doi_authors.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""Keywords linked to citations."""
```
Comment (dmlb2000) [inline metadata/orm/doi_authors.py]: Fix the comment here...

File: metadata/orm/doi_info.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    """
+    TransactionKeyValue attributes are foreign keys.
+
+    Attributes:
+        +-------------------+-------------------------------------+
+        | Name              | Description                         |
+        +===================+=====================================+
+        | doi               | Link to the DOIEntries model        |
+        +-------------------+-------------------------------------+
+        | key               | Link to the Keys model              |
```
Comment (dmlb2000) [inline metadata/orm/doi_info.py]: These aren't links... should they be?

Comment (kauberry) [reply metadata/orm/doi_info.py]: Nope. Just missed renaming the stuff in the comments. =)

File: metadata/orm/doi_info.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+from peewee import ForeignKeyField, CharField
+from metadata.orm.base import DB
+from metadata.orm.doi_entries import DOIEntries
+from metadata.rest.orm import CherryPyAPI
+from metadata.orm.utils import unicode_type
+
+
+class DOIInfo(CherryPyAPI):
+    """
+    TransactionKeyValue attributes are foreign keys.
```
[the remark this answers is not in the document]
Comment (kauberry) [reply metadata/orm/doi_info.py]: Check.

File: metadata/orm/doi_info.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""TransactionKeyValue links Transactions and Keys and Values objects."""
```
[the remark this answers is not in the document]
Comment (kauberry) [reply metadata/orm/doi_info.py]: Check.

File: metadata/orm/doi_authors.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+from peewee import CharField, IntegerField, ForeignKeyField
+from metadata.rest.orm import CherryPyAPI
+from metadata.orm.doi_entries import DOIEntries
+from metadata.orm.utils import unicode_type
+
+
+class DOIAuthors(CherryPyAPI):
```
[the remark this answers is not in the document]
Comment (dmlb2000) [reply metadata/orm/doi_authors.py]: Did you want to do the de-normalized secondary Authors table this time around or not? Should we resolve this conversation?

File: metadata/orm/doi_authors.py (hunk 3 above)
Comment (dmlb2000) [reply metadata/orm/doi_authors.py]: This doc fix still needs to be addressed.

**Comment by kauberry** [comment]
I already added the mapping table to clear up some of the redundancies that might show up. (That’s doi_author_mapping)
Thanks,
Ken

Ken J. Auberry, <mailto:Kenneth.Auberry@pnnl.gov> Senior Data Scientist
Molecular Sciences Computing Facility
Environmental Molecular Sciences Laboratory <http://www.emsl.pnnl.gov/>
email: Kenneth.Auberry@pnnl.gov <mailto:Kenneth.Auberry@pnnl.gov>
tel: 509.371.6442 <tel:+15093716442>
cell: 509.845.3010 <tel:+15098453010>




On Oct 8, 2018, at 2:13 PM, David Brown <notifications@github.com<mailto:notifications@github.com>> wrote:


@dmlb2000 commented on this pull request.

________________________________

In metadata/orm/doi_authors.py<https://github.com/pacifica/pacifica-metadata/pull/156#discussion_r223498897>:

> @@ -0,0 +1,85 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""Keywords linked to citations."""
+from peewee import CharField, IntegerField, ForeignKeyField
+from metadata.rest.orm import CherryPyAPI
+from metadata.orm.doi_entries import DOIEntries
+from metadata.orm.utils import unicode_type
+
+
+class DOIAuthors(CherryPyAPI):


Did you want to do the de-normalized secondary Authors table this time around or not? Should we resolve this conversation?

File: metadata/orm/doi_authors.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (kauberry) [reply metadata/orm/doi_authors.py]: I've added this de-normalization and mapping between doi_entries and authors

**Pull request merged**